### PR TITLE
Xenofriends 

### PR DESCRIPTION
**Pull request by fenndragon**
# Description

Adds friendly Xeno spawns alongside friendshape and even a few passive xeno mobs 

---

# Changelog

:cl:
- Add: a collection of Passive Xenos
- Add: several New friendly Xeno experimental subjects much like Friend-shaped.

### CODE REVIEW

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
This looks sadly like a mess, please avoid making inventories for xenos.

If you wana make Xenos Neutral/Pet please do what you have done in line 24 to 71 with all xenos

To add the new xenos, please add them in the xeno.yml and neutral/pet version in the xenopet.yml

Theres double files of the textures, please delete the doubles.

Failed Tests are valid, this need to be resolved.

**Comment by fenndragon** [comment]
> This looks sadly like a mess, please avoid making inventories for xenos.
> 
> If you wana make Xenos Neutral/Pet please do what you have done in line 24 to 71 with all xenos
> 
> To add the new xenos, please add them in the xeno.yml and neutral/pet version in the xenopet.yml
> 
> Theres double files of the textures, please delete the doubles.
> 
> Failed Tests are valid, this need to be resolved.

have been trying to work out where the fails are, any assistance would be appreciated.

**Comment by fenndragon** [comment]
> This looks sadly like a mess, please avoid making inventories for xenos.
> 
> If you wana make Xenos Neutral/Pet please do what you have done in line 24 to 71 with all xenos
> 
> To add the new xenos, please add them in the xeno.yml and neutral/pet version in the xenopet.yml
> 
> Theres double files of the textures, please delete the doubles.
> 
> Failed Tests are valid, this need to be resolved.

also the new/duplicated sprite (patriarch) is a placeholder to go ahead and set up the file path when I choose to actually alter his sprite. (was thinking of some scars or markings)

**Comment by DakoDemon** [comment]
Gonna give this a look-see to see what's throwing errors. I did some of the code as well.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      path: /Audio/Animals/lizard_happy.ogg
+
+- type: entity
+  name: FXE Subject 7355
+  parent: MobXenoNeutralRavager
+  id: MobXenoSubjectTess
+  description: An extremely oddly coloured xeno, with glowing stripes. An odd mutation.
+  components:
+  - type: NpcFactionMember
+     factions:
```
[the remark this answers is not in the document]
Comment (fenndragon) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml]: And fixed.

**Comment by fenndragon** [comment]
> This looks sadly like a mess, please avoid making inventories for xenos.
> 
> If you wana make Xenos Neutral/Pet please do what you have done in line 24 to 71 with all xenos
> 
> To add the new xenos, please add them in the xeno.yml and neutral/pet version in the xenopet.yml
> 
> Theres double files of the textures, please delete the doubles.
> 
> Failed Tests are valid, this need to be resolved.

Inventories for specifically the Xenos that work for CC are important to the player as a ghost role. usually the inventory and such is added via admin commands, this is just simplifying the process for the admin to create the ghost role. 

there are a few Xenos whom have not been added as a neutral entity either due to lack of intelligence or the concept that they are inherently aggressive, such as the queen. 

Generally I would assume neutral Xenos would be present in the absence of an active hive or queen to organize them.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    displayName: ID
+  - name: pocket1
+    slotTexture: pocket
+    slotFlags: POCKET
+    slotGroup: MainHotbar
+    stripTime: 3
+    uiWindowPos: 0,3
+    strippingWindowPos: 0,4
+    displayName: Pocket 1
+    stripHidden: false
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:106]:
```suggestion
```

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        path: /Audio/Weapons/Xeno/alien_claw_flesh3.ogg
+    angle: 0
+    animation: WeaponArcClaw
+    damage:
+      types:
+        Blunt: 5
+        Slash: 5
+  - type: Puller
+    needsHands: false
+  - type: GhostTakeoverAvailable
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:22]: MobSpaceFriend is only used two, you could just merge those those with MobXenoNeutralRouny

and parent MobXenoFriendShaped to MobXenoNeutralRouny

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+  - type: MovementAlwaysTouching
+  - type: MeleeWeapon
+    soundHit:
+        path: /Audio/Weapons/Xeno/alien_claw_flesh3.ogg
+    angle: 0
+    animation: WeaponArcClaw
+    damage:
+      types:
+        Blunt: 5
+        Slash: 5
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:19]:
```suggestion
```
Avoid changing Xeno Default Attacks

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+  components:
+  - type: NpcFactionMember
+    factions:
+    - PetsNT
+  - type: Sprite
+    drawdepth: Mobs
+    sprite: Mobs/Aliens/Xenos/rouny.rsi
+    scale: 0.7, 0.7
+  - type: Inventory
+    speciesId: rouny
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:122]:
```suggestion
```
This is not needed.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+  id: MobXenoFriendShaped
+  description: A very clearly friend-shaped Xeno.
+  components:
+  - type: NpcFactionMember
+    factions:
+    - PetsNT
+  - type: Sprite
+    drawdepth: Mobs
+    sprite: Mobs/Aliens/Xenos/rouny.rsi
+    scale: 0.7, 0.7
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:120]: If parent already has rouny sprite, this is not needed.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: patriarch
+      Critical:
+        Base: patriarch_crit
+      Dead:
+        Base: patriarch_dead
+  - type: Inventory
+    speciesId: praetorian
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:169]:
```suggestion
```
This is not needed

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+  name: FXES
+  parent: MobXenoNeutralDrone
+  id: MobXenoFXES
+  description: A very clearly friend-shaped Xeno.
+  components:
+  - type: NpcFactionMember
+    factions:
+    - PetsNT
+  - type: Inventory
+    speciesId: drone
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:200]:
```suggestion
```
This is not needed

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+  name: Hell-Shaped
+  parent: MobXenoNeutralRavager
+  id: MobXenoHellShaped
+  description: A very clearly friend-shaped Xeno.
+  components:
+  - type: NpcFactionMember
+    factions:
+    - PetsNT
+  - type: Inventory
+    speciesId: ravager
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:233]:
```suggestion
```
This is not needed

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: subject7355
+      Critical:
+        Base: subject7355_crit
+      Dead:
+        Base: subject7355_dead
+  - type: Inventory
+    speciesId: ravager
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:287]:
```suggestion
```
This is not needed

File: patriarch.rsi/meta.json
Comment (FoxxoTrystan) [inline patriarch.rsi/meta.json:1]: This file needs to be removed.
Duplicate.

File: patriarch.rsi/patriarch.png
Comment (FoxxoTrystan) [inline patriarch.rsi/patriarch.png:1]: This file needs to be removed.
Duplicate.

File: patriarch.rsi/patriarch_crit.png
Comment (FoxxoTrystan) [inline patriarch.rsi/patriarch_crit.png:1]: This file needs to be removed.
Duplicate.

File: patriarch.rsi/patriarch_dead.png
Comment (FoxxoTrystan) [inline patriarch.rsi/patriarch_dead.png:1]: This file needs to be removed.
Duplicate.

File: patriarch.rsi/patriarch_sleeping.png
Comment (FoxxoTrystan) [inline patriarch.rsi/patriarch_sleeping.png:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/glow.png
Comment (FoxxoTrystan) [inline subject7355.rsi/glow.png:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/meta.json
Comment (FoxxoTrystan) [inline subject7355.rsi/meta.json:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/subject7355.png
Comment (FoxxoTrystan) [inline subject7355.rsi/subject7355.png:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/subject7355_crit.png
Comment (FoxxoTrystan) [inline subject7355.rsi/subject7355_crit.png:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/subject7355_dead.png
Comment (FoxxoTrystan) [inline subject7355.rsi/subject7355_dead.png:1]: This file needs to be removed.
Duplicate.

File: subject7355.rsi/subject7355_sleeping.png
Comment (FoxxoTrystan) [inline subject7355.rsi/subject7355_sleeping.png:1]: This file needs to be removed.
Duplicate.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+  id: MobXenoNeutralDrone
+  parent: MobXenoDroneNPC
+  description: They mostly come at night. Mostly.
+  components:
+  - type: LanguageKnowledge
+    speaks:
+    - Xeno
+    understands:
+    - Xeno
+    - GalacticCommon
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:69]: If neureal, is this one missing a faction?

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Remove Duplicates files.

Some Prototypes has Puller and some does not, is this normal?
If you want all xenos to have puller, i suggest to edit the XenoBase Prototype.

MobSpaceFriend can be merged with MobXenoNeutralRouny.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml (hunk 3 above)
Comment (fenndragon) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:22]: I found this easier as it was less code to just specify the difference in faction than it was to specify a different sprite and handle all the differences (combat values and such) created by having it parented to something that's own parent is in fact the MobXenoRounyNPC

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml (hunk 4 above)
Comment (fenndragon) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:19]: this is not a default xeno.  therefore it is not a default xeno attack. it's intended to be a possible ghost role. therefore it's tailored to be used by a player. I've strived to try and avoid changing values and parameters on the actual basegame xeno files so long as it did not involve fixing a bug (like the unrevivable bug)

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml (hunk 6 above)
Comment (fenndragon) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:111]: I specified it so I could choose a rescale, I am uncertain how to give the scale change for friend-shaped without this.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml (hunk 11 above)
Comment (fenndragon) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:69]: thanks for catching this, I didn't realize I had forgotten to give this one the passive faction.

**Comment by fenndragon** [comment]
> Remove Duplicates files.
> 
> Some Prototypes has Puller and some does not, is this normal? If you want all xenos to have puller, i suggest to edit the XenoBase Prototype.
> 
> MobSpaceFriend can be merged with MobXenoNeutralRouny.

do you think I should make that change to the base xeno? I have avoided changing much in relation to the base xeno file- but I suppose it is less code to do that and it doesn't really effect the npc Xenos too much

**Comment by fenndragon** [comment]
pretty sure that's a false positive

**Comment by Mnemotechnician** [comment]
it is, test fail is fake.

**Comment by Mnemotechnician** [comment]
Also, that entire cl should certainly be compressed into
```markdown
:cl:
- add: Added several new xenomorph spawns for admins.
```

**Comment by Fansana** [comment]
I haven't looked much at this but I want to add my 5 cents.
Xenos should not have pockets.
Xenos should not have hands
Xenos should be able to drag with their tails
Passive Xenos should not spawn naturally.
Passive Xenos should look different from aggressive Xenos. Either by size or color.

**Comment by DakoDemon** [comment]
> I haven't looked much at this but I want to add my 5 cents. Xenos should not have pockets. Xenos should not have hands Xenos should be able to drag with their tails Passive Xenos should not spawn naturally. Passive Xenos should look different from aggressive Xenos. Either by size or color.

Sure! I'll go through and help change some of these. I did some of the code changes myself, cause I wanted to add actual like NT Xeno Pets that can be spawned in.
I do understand the pocket thing, I just wanted to give passive Xenos a bit more utility.
The hand was supposed to be a kinda, they're holding it with their mouths type thing, but we can take that out if absolutely neccesary.
Tail-drag was supposed to kinda replace the need for a hand, so definitely agree that that should be kept in.
The only things I'd say should spawn 'naturally' are the pet varients, if ANYTHING, otherwise, yeah, Admin spawn is the easiest.
As for the looks, we CAN go through and add some changes to the sprites if need be, or make them smaller, but we were going for the idea that these are Xenos that have broken off from the Queen and are now their own entity, so they'd still be mostly the same. But yeah! Fenn and I can do some changes. Maybe desaturate their colours some.

File: Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml (hunk 4 above)
Comment (FoxxoTrystan) [reply Resources/Prototypes/Floof/Entities/Mobs/NPCs/xenopet.yml:19]: MeleeWeapon Prototype is already loaded by MobXenoRounyNPC, thats load RunnerNPC with the damage prototype.

This is again, undeeded.

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
I feel InteractionPopup could just be on the default xeno base but il let this go.

**Comment by fenndragon** [comment]
> I haven't looked much at this but I want to add my 5 cents. Xenos should not have pockets. Xenos should not have hands Xenos should be able to drag with their tails Passive Xenos should not spawn naturally. Passive Xenos should look different from aggressive Xenos. Either by size or color.

I added a colored point light to each of the neutral Xenos to differentiate them from the regular. 

They don't have pockets
They don't have hands
They all have tail puller
they do not spawn naturally.

**Comment by fenndragon** [comment]
weapon thing removed now

**Comment by fenndragon** [comment]
ok it's given me the same failed test every time I've made a change, maybe that's not a false positive? what could be causing that

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Code look overall good now heres things to fix those before merge.

> Changlog need to be shorter please, more simple like (Added a bunch of new Xenomorph) ect....
> PointLight While being very tiny, im not sure that will let peoples know whos friend or not, tho i will leave this to be not approved by me but Floof/Memeji.
Tho i would recomend to do pehaps some sprites changes, pehaps.

**Comment by fenndragon** [comment]
> Code look overall good now heres things to fix those before merge.
> 
> > Changlog need to be shorter please, more simple like (Added a bunch of new Xenomorph) ect....
> > PointLight While being very tiny, im not sure that will let peoples know whos friend or not, tho i will leave this to be not approved by me but Floof/Memeji.
> > Tho i would recomend to do pehaps some sprites changes, pehaps.

well, from my experience with pointlight it tints the color of the sprite. which is why I chose to use it as a quickfix for now to allow time for the new sprites to be made. 

as for the rest, I'll get set on the changelog

**Comment by fenndragon** [comment]
> Code look overall good now heres things to fix those before merge.
> 
> > Changlog need to be shorter please, more simple like (Added a bunch of new Xenomorph) ect....
> > PointLight While being very tiny, im not sure that will let peoples know whos friend or not, tho i will leave this to be not approved by me but Floof/Memeji.
> > Tho i would recomend to do pehaps some sprites changes, pehaps.

alright done~

**Review by Fansana — APPROVED** [review]

**Review by Memeji — APPROVED** [review]

**Pull request merged**